### DISCUSSION TITLE
Inefficient extension step with highish number of reads

### DISCUSSION THREAD

**Discussion by shihanli92** [liulab-dfci/TRUST4#40] [General]
Hey, 

Thanks for making this great tool. I was testing TRUST4 for its ability to detect TCR clones from deep sequencing data where the TCRs have been enriched to see how it would compare with miXCR. I applied it to a paired fastq file with ~300k reads and found that the time required extension step seemed to increase heavily as the number of reads increased. 

Here's the plot of what I mean. Not sure if this is something to do on my end or something worth thinking about in the future. 

![image](https://user-images.githubusercontent.com/60378562/118447228-df26a900-b733-11eb-9e55-eeb4983f67fd.png)

All the best,
Tony

**Comment by mourisl** [comment]
Thanks for the testing. Indeed this is too slow. Could you please show me the running log output on the screen? For TCR/BCR enriched data, you may want to try the option "--repseq". Thank you.

**Reply by shihanli92** [reply]
Ah cool, I missed the repseq option, will see if that improves the timing. Thanks for that
